pkg/report/linux.go

```diff
@@ -661,3 +711,93 @@ var linuxOopses = []*oops{
 		[]*regexp.Regexp{},
 	},
 }
+
+func isCorrupted(desc string, text string) bool {
+	if !strings.Contains(text, "Call Trace") && !strings.Contains(text, "backtrace") {
+		// Text must contain 'Call Trace' or 'backtrace'.
+		return true
+	}
+	corruptedDescRegExps := []*regexp.Regexp{
+		// 'kernel panic: Fatal exception' is usually printed after BUG,
+		// so if we captured it as a report description, that means the
+		// report got truncated and we missed the actual BUG header.
+		regexp.MustCompile(`kernel panic: Fatal exception`),
+		// Same, but for WARNINGs and KASAN reports.
+		regexp.MustCompile(`kernel panic: panic_on_warn set`),
+		// Sometimes timestamps get merged into the middle of report description.
+		regexp.MustCompile(`\[ *[0-9]+\.[0-9]+\]`),
+		regexp.MustCompile(`\[ *[0-9]+\.NUM\]`),
+		regexp.MustCompile(`\[ *NUM\.NUM\]`),
+		regexp.MustCompile(`\[ *NUM\.[0-9]+\]`),
+	}
+	for _, re := range corruptedDescRegExps {
+		if re.MatchString(desc) {
+			return true
+		}
+	}
+	corruptedDescStrings := []string{
+		// Sometimes with such BUG failures, the second part of the header doesn't get printed
+		// or gets corrupted, because kernel prints it as two separate printk() calls.
+		"BUG: unable to handle kernel",
+		// If we failed to extract function name where the fault happened,
+		// the report is most likely truncated.
+		"general protection fault",
+		"BUG: bad unlock balance",
+		"divide error",
+		"invalid opcode",
+	}
+	for _, s := range corruptedDescStrings {
+		if strings.TrimSpace(desc) == s {
+			return true
+		}
+	}
+	corruptedTextRegExps := []*regexp.Regexp{
+		// If report text contains 'printk messages dropped' it is most likely corrupted.
+		regexp.MustCompile(`printk messages dropped`),
+	}
+	for _, re := range corruptedTextRegExps {
+		if re.MatchString(text) {
+			return true
+		}
+	}
+	crashTypes := []string{"BUG", "WARNING", "INFO", "KASAN", "KMSAN", "UBSAN"}
+	for _, crash := range crashTypes {
+		// If description contains 'BUG', 'WARNING', etc,
+		// text must also contain it.
+		if strings.Contains(desc, crash) && !strings.Contains(text, crash) {
+			return true
+		}
+	}
+	if strings.HasPrefix(desc, "possible deadlock") {
+		// For 'possible deadlock' reports lets use 'unsafe locking scenario'
+		// string in text as a signal whether the report got truncated.
+		if !strings.Contains(text, "unsafe locking scenario") {
+			return true
+		}
+	}
+	if strings.HasPrefix(desc, "KASAN") {
+		// For KASAN reports lets use 'Allocated' and 'Freed' as signals.
+		if !strings.Contains(text, "Allocated") {
```

Comment:
This should be a filed of oopsFormat. I.e. if we match this regexp, these patterns must follow in report. If not, the report is corrupted.

Reply:
The regexps in `oopsFormat` are more about report title and this is about report body. I could add another regexp to `oopsFormat` that should match the whole report.